common/src/main/scala/org/neo4j/spark/reader/BasePartitionReader.scala

```diff
@@ -29,18 +29,22 @@ abstract class BasePartitionReader(private val options: Neo4jOptions,
   protected val driverCache: DriverCache = new DriverCache(options.connection, name)
 
   private val query: String = new Neo4jQueryService(options,
-    new Neo4jQueryReadStrategy(filters, partitionSkipLimit, requiredColumns.fieldNames, aggregateColumns))
+    new Neo4jQueryReadStrategy(filters, partitionSkipLimit, requiredColumns.fieldNames, aggregateColumns, jobId))
     .createQuery()
 
   private var nextRow: InternalRow = _
 
   private lazy val values = {
-    val params = mutable.HashMap[String, Any]()
+    val params = new java.util.HashMap[String, Any]()
     params.put(Neo4jQueryStrategy.VARIABLE_SCRIPT_RESULT, scriptResult)
```

Comment:
do this line and the two below make sense in the context of GDS?

Reply:
Yes the `params`, in the case of GDS, are meant as input parameters

Reply:
Is not **strictly** related to the GDS, I just changed from mutable avoiding unnecessary conversion between Scala/Java maps, as the unflattened map that we have here:
https://github.com/neo4j-contrib/neo4j-spark-connector/pull/506/files#diff-d1bb03b0e08a939701549814cf7d852d56a95d6401526c8252627b824845f606R44
is already a Java map